sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample6_PublishEventsWithCustomMetadata.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Samples.Infrastructure;
+
+namespace Azure.Messaging.EventHubs.Samples
+{
+    /// <summary>
+    ///   An example of publishing events, extending the event data with custom metadata.
+    /// </summary>
+    ///
+    public class Sample6_PublishEventsWithCustomMetadata : ISample
+    {
+        /// <summary>
+        ///   The name of the sample.
+        /// </summary>
+        ///
+        public string Name { get; } = nameof(Sample6_PublishEventsWithCustomMetadata);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description { get; } = "An example of publishing events, extending the event data with custom metadata.";
+
+        /// <summary>
+        ///   Runs the sample using the specified Event Hubs connection information.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string for the Event Hubs namespace that the sample should target.</param>
+        /// <param name="eventHubName">The name of the Event Hub, sometimes known as its path, that she sample should run against.</param>
+        ///
+        public async Task RunAsync(string connectionString,
+                                   string eventHubName)
+        {
+            // We will start by creating a client and a producer, each using their default set of options.
+
+            await using (var client = new EventHubClient(connectionString, eventHubName))
+            await using (EventHubProducer producer = client.CreateProducer())
+            {
+                // Because an event consists mainly of an opaque set of bytes, it may be difficult for consumers of those events to
+                // make informed decisions about how to process them.
+                //
+                // In order to allow event publishers to offer better context for consumers, event data may also contain custom metadata,
```

Comment:
We should not be calling a concept "metadata" in  the description but then "properties" in the APIs.